src/preamble.js

```diff
@@ -122,6 +122,11 @@ function assert(condition, text) {
 
 var globalScope = this;
 
+Module['runtimeInitializedPromise'] = Promise.resolve();
```

Comment:
Ah right, meant to ask about this. Either way seems fine to me, but I did it this way initially for two reasons:

1. if `deferRuntimeInitialization` is something that may potentially be used throughout the code going forward, I'd imagine that we'd want it to apply to `onRuntimeInitialized` in addition to just this promise, in which case the most obvious implementation to me was to always keep this here and wait for it to resolve before running `onRuntimeInitialized`.

2. For my wasm libraries, I just made every public field a Promise and every method return a Promise, but I could imagine that others may want the option to keep their APIs synchronous and just expose an "is initialized" Promise (so `await lib.ready(); const foo = lib.bar();` vs `const foo = await lib.bar();`).

But I'm not sure if there are any minimum browser requirements in play here that would preclude depending on native Promise support.